geomesa-spark-jts/src/main/scala/org/apache/spark/sql/jts/SQLTypes.scala

```diff
@@ -0,0 +1,128 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.apache.spark.sql.jts
+
+import com.vividsolutions.jts.geom._
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.types._
+import org.geotools.factory.CommonFactoryFinder
+import org.geotools.geometry.jts.JTSFactoryFinder
+import org.locationtech.geomesa.spark._
+import org.opengis.filter.FilterFactory2
+
+import scala.reflect.ClassTag
+
+
+object SQLTypes {
```

Comment:
Foreshadowing: I'll make arguments below for renaming this. Not sure what the right name is.... `SpatialTypes`?